src/main/java/Duke.java

```diff
@@ -1,10 +1,193 @@
+                    System.out.print("[ ] ");
+                    System.out.println(description);
+                    break;
+
+                case ("todo"): //Add to list
+                    printLine();
+
+                    try{
+                        description = getStringFromList(inputList, 1, inputList.length);
+                    } catch (IllegalArgumentException e){
+                        System.out.println("Task cannot be empty!");
+                        break;
+                    }
+
+                    //Printing result
+                    System.out.println("Added:");
+                    System.out.println(" [T][ ] " + description);
+                    taskList.addToDo(description);
+                    System.out.println("Now you have " + taskList.getSize() + " tasks in the list.");
+                    break;
+
+                case ("deadline"):
+                    printLine();
+                    int byPosition = 0;
+                    String dueDate;
+                    for (int i=0; i<inputList.length; i++){
+                        if (inputList[i].equals("by")) {
+                            byPosition = i;
+                            break;
+                        }
+                    }
+                    
+                    // Handle invalid input
+                    try {
+                        description = getStringFromList(inputList, 1, byPosition);
+                        dueDate = getStringFromList(inputList, byPosition+1, inputList.length);
+                    } catch (IllegalArgumentException e){
+                        System.out.println("Invalid input: task/date not given!");
+                        break;
+                    }
+
+                    //Printing result
+                    System.out.println("Added:");
+                    System.out.println(" [D][ ] " + description + " (by: " + dueDate + ")");
+                    taskList.addDeadline(description, dueDate);
+                    System.out.println("Now you have " + taskList.getSize() + " tasks in the list.");
+                    break;
+
+                case ("event"):
+                    printLine();
+                    int atPosition = 0;
+                    String dateTime;
+                    for (int i=0; i<inputList.length; i++){
+                        if (inputList[i].equals("at")){
+                            atPosition = i;
+                            break;
+                        }
+                    }
+
+                    //Handle invalid input
+                    try{
+                        description = getStringFromList(inputList, 1, atPosition);
+                        dateTime = getStringFromList(inputList, atPosition+1, inputList.length);
+                    } catch (IllegalArgumentException e){
+                        System.out.println("Invalid input: task/date not given!");
+                        break;
+                    }
+
```

Comment:
In such situations where a particular feature may require extensive code to achieve, it may be better for readibility to abstract the relevant blocks of code to separate functions (SLAP)